src/rp/index.ts

```diff
@@ -166,29 +171,37 @@ export class RelyingPartySolution {
   /**
    * Compose and send an Authorization Response in the context of an authorization request flow.
    *
-   * @todo MUST encrypt response payload
    * @todo MUST add presentation_submission
    *
    * @param requestObj The incoming request object, which the requirements for the requested authorization
    * @param vp_token The signed Verified Presentation token with data to send.
+   * @param entity The RP entity configuration
    * @returns The response from the RP
-   * @throws IoWalletError if the submission fails.
+   * @throws {IoWalletError} if the submission fails.
+   * @throws {NoSuitableKeysFoundInEntityConfiguration} If entity do not contain any public key
    *
    */
   async sendAuthorizationResponse(
     requestObj: RequestObject,
-    vp_token: string
+    vp_token: string,
+    entity: RpEntityConfiguration
   ): Promise<string> {
     // the request is an unsigned jws without iss, aud, exp
     // https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#name-signed-and-encrypted-respon
-    // TODO: [SIW-351] MUST be encrypted
-    const authzResponsePayload = {
+    const jwk = this.choosePublicKeyToEncrypt(entity);
+
+    const authzResponsePayload = JSON.stringify({
       state: requestObj.payload.state,
       // TODO: [SIW-352] MUST add presentation_submission
       // presentation_submission:
       vp_token,
-    };
-    const formBody = new URLSearchParams({ response: authzResponse });
+    });
+    const encrypted = await new EncryptJwe(authzResponsePayload, {
+      alg: jwk.alg,
+      enc: "A256CBC-HS512",
```

Comment:
alg is related to enc. With `A256CBC-HS512` you can use only `RSA-OAEP-256`. For this reason it is perhaps better to pass the algorithm to use to `choosePublicKeyToEncrypt`

Reply:
Do we have to consider jwk on this choice? I'd rather use a fixed algorithm

Reply:
You can use a fixed algorithm, the important thing is that JWK is compatible with the algorithm used.
With `RSA-OAEP-256` you can only use `A256CBC-HS512` and with  `RSA-OAEP` you can only use `A128CBC-HS256`.

in your code you fixed the `enc` at `A256CBC-HS512` but your JWK can be either of type `RSA-OAEP-256` and `RSA-OAEP`

Reply:
my suggestion is to fix the key algorithm to `RSA-OAEP-256` with `A256CBC-HS512`. If then the key is not found then switch to `RSA-OAEP` with `A128CBC-HS256`. If that isn't found either, then an error returns

Reply:
done in https://github.com/pagopa/io-react-native-wallet/pull/21/commits/8316b4fe3c23d8b236991ed8fa26f0ddab12ba5b